### ISSUE TITLE
Experiments from the paper: Multivariate Convolutional Sparse Coding for Electromagnetic Brain Signals 

### ISSUE THREAD

**Issue by dinalzein** [alphacsc/alphacsc#76]
I'm trying to replicate some experiments from the paper: "Multivariate Convolutional Sparse Coding for Electromagnetic Brain Signals " (mainly figure 3) but I'm not sure how is GreedyCDL supported for univariate signals (is't only by forcing `rank1=False `and `uv_constraint='auto'` )? 
Also, for simulating the data, are you using `load_data` in [simulate.py](https://github.com/alphacsc/alphacsc/blob/master/alphacsc/datasets/simulate.py) or is there another function that simulates data for multiple channels?

**Comment by agramfort** [comment]
with only 1 channel there is no Rank 1 constraint required. See our
examples with univariate simulations eg
https://alphacsc.github.io/auto_examples/noise_alpha/plot_simulate_alphacsc.html#sphx-glr-auto-examples-noise-alpha-plot-simulate-alphacsc-py

now Fig 3 in the paper your mention varies P ie the number of channels so
it's not univariate in this figure.

**Comment by dinalzein** [comment]
Yes, but I wanted to compare the vanilla CSC with GreedyCSC algorithm on univariate signal. I tried this by having `rank1=False`, `uv_constraint='auto'` , and `solver_d = 'fista'`; it worked. 

In the paper, there is no comparison between with and without the rank1 constraint on the multivariate signal. I want to do this experiment but I'm getting the following attached error when having` rank1=False`, `uv_constraint='auto'`, and`solver_d = 'fista'` on the mne dataset. 

<img width="1023" alt="error" src="https://user-images.githubusercontent.com/44267423/159561853-2beadf0d-20f3-4329-acd4-b10604d61718.png">



The error seems to come from line 39 `@numba.jit((numba.float64[:, :, :],), nopython=True, cache=True)` in [compute_constants.py](https://github.com/alphacsc/alphacsc/blob/master/alphacsc/utils/compute_constants.py)

**Comment by agramfort** [comment]
please share a self-contained code snippet to replicate the problem
